src/main/scala/com/microsoft/hyperspace/index/factories.scala

```diff
@@ -20,31 +20,32 @@ import org.apache.hadoop.conf.Configuration
 import org.apache.hadoop.fs.{FileSystem, Path}
 
 trait IndexLogManagerFactory {
-  def create(indexPath: Path): IndexLogManager
+  def create(indexPath: Path, configuration: Configuration = new Configuration): IndexLogManager
 }
 
 object IndexLogManagerFactoryImpl extends IndexLogManagerFactory {
-  override def create(indexPath: Path): IndexLogManager = {
-    new IndexLogManagerImpl(indexPath)
+  override def create(indexPath: Path,
+      configuration: Configuration = new Configuration): IndexLogManager = {
+    new IndexLogManagerImpl(indexPath, configuration)
   }
 }
 
 trait IndexDataManagerFactory {
-  def create(indexPath: Path): IndexDataManager
+  def create(indexPath: Path, configuration: Configuration = new Configuration): IndexDataManager
 }
 
 object IndexDataManagerFactoryImpl extends IndexDataManagerFactory {
-  override def create(indexPath: Path): IndexDataManager = {
-    new IndexDataManagerImpl(indexPath)
+  override def create(indexPath: Path,
+      configuration: Configuration = new Configuration): IndexDataManager = {
+    new IndexDataManagerImpl(indexPath, configuration)
   }
 }
 
 trait FileSystemFactory {
-  def create(path: Path): FileSystem
+  def create(path: Path, configuration: Configuration = new Configuration): FileSystem
```

Comment:
Can we remove default parameter value from `trait`? (I am not a big fan of it since derived class can override the value and can cause some confusion).

Reply:
👍